### ISSUE TITLE
Test problems in current parallel test-run

### ISSUE THREAD

**Issue by bigbes** [tarantool/tarantool#2436]
### `replication/autobootstrap.test.lua`, tag: `vinyl` (sequencial run (with `-j`), probably some race condition) fails sometimes

log:
```
[023] replication/autobootstrap.test.lua              memtx           [ pass ]
[023] replication/autobootstrap.test.lua              vinyl
[023]
[023] [Instance "autobootstrap3" returns with non-zero exit code: 1]
[023]
[023] Last 15 lines of Tarantool Log file [Instance "autobootstrap3"][/Users/blikh/src/work/tarantool/test/var/023_replication/autobootstrap3.log]:
[023] 2017-05-16 16:50:18.366 [6943] main/101/tarantoolctl I> mapping 268435456 bytes for tuple arena...
[023] 2017-05-16 16:50:18.373 [6943] iproto/101/main I> binary: bound to unix/:/Users/blikh/src/work/tarantool/test/var/023_replication/autobootstrap3.sock
[023] 2017-05-16 16:50:18.373 [6943] main/105/applier/cluster@unix/:/Users/ I> remote master is 1.7.4 at unix/:/Users/blikh/src/work/tarantool/test/var/023_replication/autobootstrap1.sock
[023]
[023] 2017-05-16 16:50:18.373 [6943] main/106/applier/cluster@unix/:/Users/ I> remote master is 1.7.4 at unix/:/Users/blikh/src/work/tarantool/test/var/023_replication/autobootstrap2.sock
[023]
[023] 2017-05-16 16:50:18.373 [6943] main/107/applier/cluster@unix/:/Users/ I> remote master is 1.7.4 at unix/:/Users/blikh/src/work/tarantool/test/var/023_replication/autobootstrap3.sock
[023]
[023] 2017-05-16 16:50:18.374 [6943] main/106/applier/cluster@unix/:/Users/ I> failed to authenticate
[023] 2017-05-16 16:50:18.374 [6943] main/106/applier/cluster@unix/:/Users/ xrow.cc:431 E> ER_LOADING: Instance bootstrap hasn't finished yet
[023] 2017-05-16 16:50:19.374 [6943] main/106/applier/cluster@unix/:/Users/ I> failed to authenticate
[023] 2017-05-16 16:50:19.374 [6943] main/106/applier/cluster@unix/:/Users/ xrow.cc:431 E> ER_NO_SUCH_USER: User 'cluster' is not found
[023] 2017-05-16 16:50:19.375 [6943] main/101/tarantoolctl xrow.cc:431 E> ER_NO_SUCH_USER: User 'cluster' is not found
[023] 2017-05-16 16:50:19.375 [6943] main/101/tarantoolctl F> can't initialize storage: User 'cluster' is not found
[023] 2017-05-16 16:50:19.375 [6943] main/101/tarantoolctl F> can't initialize storage: User 'cluster' is not found
[023] [ fail ]
```
reproduce file (sequential):
```
- [replication/autobootstrap.test.lua, memtx]
- [replication/autobootstrap.test.lua, vinyl]
```

### `app/socket.test.lua` (only on Ubuntu 14.04.1 on sh1)

```
[033] app/socket.test.lua                                             [ fail ]
[033]
[033] Test failed! Result content mismatch:
[033] --- app/socket.result	Wed May 17 11:29:34 2017
[033] +++ app/socket.reject	Wed May 17 12:21:37 2017
[033] @@ -666,7 +666,7 @@
[033]  ...
[033]  wrong_addr == nil or #wrong_addr == 0
[033]  ---
[033] -- true
[033] +- false
[033]  ...
[033]  sc = socket('PF_INET', 'SOCK_STREAM', 'tcp')
[033]  ---
[033]
```

### `028_wal_off.list.yaml` 

```
 % cat 028_wal_off.list.yaml
- [wal_off/wal_mode.test.lua, null]
- [wal_off/oom.test.lua, null]
```

Very long diff

### `vinyl/errinj.test.lua` - sometimes (in parallel/sequential)

```
[048] vinyl/errinj.test.lua                                           [ fail ]
[048]
[048] Test failed! Result content mismatch:
[048] --- vinyl/errinj.result	Wed May 17 11:29:34 2017
[048] +++ vinyl/errinj.reject	Wed May 17 13:45:38 2017
[048] @@ -117,7 +117,7 @@
[048]  -- fails due to scheduler timeout
[048]  box.snapshot();
[048]  ---
[048] -- error: Error injection 'vinyl dump'
[048] +- ok
[048]  ...
[048]  fiber.sleep(0.06);
[048]  ---
[048]
[048] Last 15 lines of Tarantool Log file [Instance "vinyl"][/home/taransible/tarantool/test/var/048_vinyl/vinyl.log]:
[048] 2017-05-17 13:45:37.559 [30383] main/209/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.561 [30383] main/210/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.562 [30383] main/211/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.564 [30383] main/212/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.565 [30383] main/213/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.567 [30383] main/214/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.568 [30383] main/215/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.569 [30383] main/216/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.571 [30383] main/217/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.573 [30383] main/218/lua txn.cc:218 E> ER_WAL_IO: Failed to write to disk
[048] 2017-05-17 13:45:37.732 [30383] main/102/vinyl.scheduler I> 520/0: dump started
[048] 2017-05-17 13:45:37.733 [30383] snapshot/101/main I> saving snapshot `/home/taransible/tarantool/test/var/048_vinyl/vinyl/00000000000000001058.snap.inprogress'
[048] 2017-05-17 13:45:37.735 [30383] snapshot/101/main I> done
[048] 2017-05-17 13:45:38.311 [30383] main/102/vinyl.scheduler I> 520/0: dump completed
[048] 2017-05-17 13:45:38.556 [30383] main/226/lua I> vinyl checkpoint done
```

```
 % cat /home/taransible/tarantool/test/var/reproduce/048_vinyl.list.yaml
- [vinyl/info.test.lua, null]
- [vinyl/errinj.test.lua, null]
```

### `wal_off/snapshot_stress.test.lua` - sometimes, standalone run in parallel test-run

```
[022] wal_off/snapshot_stress.test.lua                                [ fail ]
[022]
[022] Test failed! Result content mismatch:
[022] --- wal_off/snapshot_stress.result        Wed May 17 11:29:34 2017
[022] +++ wal_off/snapshot_stress.reject        Wed May 17 14:05:39 2017
[022] @@ -402,7 +402,7 @@
[022]  ...
[022]  snapshot_check_failed;
[022]  ---
[022] -- false
[022] +- true
[022]  ...
[022]  log.info('Part II: checking snapshot done');
[022]  ---
[022]
[022] Last 15 lines of Tarantool Log file [Instance "wal"][/home/taransible/tarantool/test/var/022_wal_off/wal.log]:
[022] 2017-05-17 14:05:36.615 [31712] snapshot/101/main I> saving snapshot `/home/taransible/tarantool/test/var/022_wal_off/wal/00000000
000000010414.snap.inprogress'
[022] 2017-05-17 14:05:36.630 [31712] snapshot/101/main I> done
[022] 2017-05-17 14:05:36.995 [31712] main/190/lua I> vinyl checkpoint done
[022] 2017-05-17 14:05:37.044 [31712] snapshot/101/main I> saving snapshot `/home/taransible/tarantool/test/var/022_wal_off/wal/00000000
000000012734.snap.inprogress'
[022] 2017-05-17 14:05:37.057 [31712] snapshot/101/main I> done
[022] 2017-05-17 14:05:37.577 [31712] main/190/lua I> vinyl checkpoint done
[022] 2017-05-17 14:05:37.613 [31712] snapshot/101/main I> saving snapshot `/home/taransible/tarantool/test/var/022_wal_off/wal/00000000
000000015054.snap.inprogress'
[022] 2017-05-17 14:05:37.629 [31712] snapshot/101/main I> done
[022] 2017-05-17 14:05:38.032 [31712] main/190/lua I> vinyl checkpoint done
[022] 2017-05-17 14:05:38.071 [31712] snapshot/101/main I> saving snapshot `/home/taransible/tarantool/test/var/022_wal_off/wal/00000000
000000017374.snap.inprogress'
[022] 2017-05-17 14:05:38.085 [31712] snapshot/101/main I> done
[022] 2017-05-17 14:05:38.399 [31712] main/190/lua I> vinyl checkpoint done
[022] 2017-05-17 14:05:38.440 [31712] main/107/console/::1:23180 I> Part I: creating snapshot done
[022] 2017-05-17 14:05:38.627 [31712] main/107/console/::1:23180 I> Part II: checking snapshot start
[022] 2017-05-17 14:05:39.497 [31712] main/107/console/::1:23180 I> Part II: checking snapshot done
```

```
 % cat var/reproduce/022_wal_off.list.yaml
- [wal_off/snapshot_stress.test.lua, null]
```

**Comment by bigbes** [comment]
Updated

**Comment by bigbes** [comment]
parallel test-run additionally can be enabled for next suites: `wal_off`, `vivnyl`, `xlog`

**Comment by rtsisyk** [comment]
> box/errinj.test.lua - sometimes, standalone

https://github.com/tarantool/tarantool/issues/2270

**Comment by rtsisyk** [comment]
> replication/autobootstrap.test.lua, tag: vinyl (sequencial run (with -j), probably some race condition) fails sometimes

I can' re-produce this problem:

    while true; do ./test-run.py autobootstrap -j; done

These tests don't fails on Travis CI too. Please find a way how to re-produce this problem and open a **separate** ticket with instructions.

**Comment by Totktonada** [comment]
@avtikhon Are all actual problems tracked in separate issues? Can we close this issue?

**Comment by ligurio** [comment]
> Are all actual problems tracked in separate issues? Can we close this issue?

@Totktonada, looks like the issue is too common. `engine/replica_join.test.lua` is still unstable, but I'll create a separate issue for it.

**Comment by Totktonada** [comment]
Sure. Thanks!
